File: clients/go-sdk/client/me.go
package client

import "context"

// MyClient is a tenant accessible to the current user.
type MyClient struct {
	ID         string `json:"id"`
	Name       string `json:"name"`
	Identifier string `json:"identifier"`
	Status     string `json:"status,omitempty"`
	CreatedAt  string `json:"createdAt,omitempty"`
	UpdatedAt  string `json:"updatedAt,omitempty"`
}

// MyClientsResponse — GET /api/me/clients.
type MyClientsResponse struct {
	Clients []MyClient `json:"clients"`
	Total   uint64     `json:"total,omitempty"`
}

// MyApplication is one application accessible within a tenant.
type MyApplication struct {
	ID            string `json:"id"`
	Code          string `json:"code"`
	Name          string `json:"name"`
	Description   string `json:"description,omitempty"`
	IconURL       string `json:"iconUrl,omitempty"`
	BaseURL       string `json:"baseUrl,omitempty"`
	Website       string `json:"website,omitempty"`
	LogoMimeType  string `json:"logoMimeType,omitempty"`
}

// MyApplicationsResponse — GET /api/me/clients/{id}/applications.
type MyApplicationsResponse struct {
	Applications []MyApplication `json:"applications"`
	Total        uint64          `json:"total,omitempty"`
	ClientID     string          `json:"clientId,omitempty"`
}

// MeResource — /api/me/*.
type MeResource struct {
	c *FlowCatalystClient
}

// Clients — GET /api/me/clients. Access is determined by user scope:
// ANCHOR → all active clients; PARTNER → IDP-granted + explicit grants;
// CLIENT → home client + IDP + explicit grants.
func (r *MeResource) Clients(ctx context.Context) (*MyClientsResponse, error) {
	var out MyClientsResponse
	if err := r.c.Get(ctx, "/api/me/clients", &out); err != nil {
		return nil, err
	}
	return &out, nil
}

// Client — GET /api/me/clients/{id}.
func (r *MeResource) Client(ctx context.Context, clientID string) (*MyClient, error) {
	var out MyClient
	if err := r.c.Get(ctx, "/api/me/clients/"+clientID, &out); err != nil {
		return nil, err
	}
	return &out, nil
}

// ClientApplications — GET /api/me/clients/{id}/applications.
func (r *MeResource) ClientApplications(ctx context.Context, clientID string) (*MyApplicationsResponse, error) {
	var out MyApplicationsResponse
	if err := r.c.Get(ctx, "/api/me/clients/"+clientID+"/applications", &out); err != nil {
		return nil, err
	}
	return &out, nil
}

File: crates/fc-sdk/src/client/me.rs
//! Current-user context (`/api/me/*`).

use super::{ClientError, FlowCatalystClient};
use serde::{Deserialize, Serialize};

/// A client accessible to the current user.
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MyClient {
    pub id: String,
    pub name: String,
    pub identifier: String,
    #[serde(default)]
    pub status: Option<String>,
    #[serde(default)]
    pub created_at: Option<String>,
    #[serde(default)]
    pub updated_at: Option<String>,
}

/// Response for listing the current user's accessible clients.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MyClientsResponse {
    pub clients: Vec<MyClient>,
    #[serde(default)]
    pub total: Option<u64>,
}

/// An application accessible to the current user within a client.
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MyApplication {
    pub id: String,
    pub code: String,
    pub name: String,
    #[serde(default)]
    pub description: Option<String>,
    #[serde(default)]
    pub icon_url: Option<String>,
    #[serde(default)]
    pub base_url: Option<String>,
    #[serde(default)]
    pub website: Option<String>,
    #[serde(default)]
    pub logo_mime_type: Option<String>,
}

/// Response for listing applications for a client.
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MyApplicationsResponse {
    pub applications: Vec<MyApplication>,
    #[serde(default)]
    pub total: Option<u64>,
    #[serde(default)]
    pub client_id: Option<String>,
}

/// Current-user accessor — created via [`FlowCatalystClient::me`].
pub struct Me<'a> {
    pub(crate) client: &'a FlowCatalystClient,
}

impl Me<'_> {
    /// All clients accessible to the current user.
    ///
    /// Access is determined by user scope:
    /// - ANCHOR: all active clients
    /// - PARTNER: IDP-granted + explicit grants
    /// - CLIENT: home client + IDP + explicit grants
    pub async fn clients(&self) -> Result<MyClientsResponse, ClientError> {
        self.client.get("/api/me/clients").await
    }

    /// A specific accessible client by ID.
    pub async fn client(&self, client_id: &str) -> Result<MyClient, ClientError> {
        self.client
            .get(&format!("/api/me/clients/{}", client_id))
            .await
    }

    /// Applications available to the current user within a client.
    pub async fn client_applications(
        &self,
        client_id: &str,
    ) -> Result<MyApplicationsResponse, ClientError> {
        self.client
            .get(&format!("/api/me/clients/{}/applications", client_id))
            .await
    }
}
